### DISCUSSION TITLE
[Bug]: WARN You (or an addon) are using the 'config' preset field. This has been replaced by 'previewAnnotations' and will be removed in 8.0

### DISCUSSION THREAD

**Discussion by The2Alert** [storybookjs/storybook#25502] [Help] [answered]
### Describe the bug

How to eliminate warning?

WARN You (or an addon) are using the 'config' preset field. This has been replaced by 'previewAnnotations' and will be removed in 8.0

### To Reproduce

<pre>storybook dev -p 6006</pre>

### System

```shell
Environment Info:

  System:
    OS: Windows 10 10.0.22621
    CPU: (44) x64 Intel(R) Xeon(R) CPU E5-2696 v4 @ 2.20GHz
  Binaries:
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.1.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1848.0), Chromium (114.0.1823.67)
  npmPackages:
    @storybook/addon-essentials: 7.0.26 => 7.0.26
    @storybook/addon-links: 7.0.26 => 7.0.26
    @storybook/blocks: 7.0.26 => 7.0.26
    @storybook/nextjs: ^7.0.26 => 7.0.26
    @storybook/react: 7.0.26 => 7.0.26
```

**Comment by apennell** [comment]
I've combed all through my code and the Storybook docs to try to find what is causing this warning for me and I can't figure it out. I found the migration docs that note the change for 7.0, but it wasn't clear what the context is and what `config` it refers to. Can more information be provided about what `config` needs replacing? Or is possible that it's an underlying addon, and therefore not something I can fix myself?

**Comment by fmagaldea** [comment]
Same warning with storybook & co version `7.3.2`.

**Comment by bazy1983** [comment, marked as answer]
i think the deprecation message can be improved a little bit. but here's what i found
in `@storybook/core-server/dist/presets/common-preset.js` file, there is a function called `previewAnnotations` that reads your main.js(ts) and try to find a property `config` either in your configuration json or in any of your addons.
if that function detects `config` property, it will fire the deprecation warning message.
in my case, the deprecation was fired because of `@storybook/addon-knobs`.
the easy way to figure out what triggers the deprecation message in your setup is to go to the `common-preset.js` in `node_modules`. search for the deprecation message. add a `console.log(config)` then run storybook. it will log out the affected addon. the more lengthy way, is by commenting out your addons one by one and run storybook.
Solutions
A- Either wait until the maintainers for the addon to upgrade their code.
B- Patch it yourself.

**Comment by shilman** [comment]
Thanks for the tip @bazy1983 !

If you find an addon that needs to be updated, we have an umbrella issue here with more info on how to go about it https://github.com/storybookjs/storybook/issues/20529

**Comment by bazy1983** [comment]
@shilman thanks for the reply.
also, i would suggest that if cli can log out the affected addons (since `config` array already has the list) with the deprecation message. it would makes a little easier for users to understand what's going on and what to look out for. i would really appreciate it!
